walkthroughs/howto-k8s-mtls-sds-based/README.md

```diff
@@ -0,0 +1,605 @@
+
+Refer to [Envoy SDS](https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret) docs for additional details on Envoy's Secret Discovery Service.
+
+SPIRE will be used as SDS provider in this walk through and will be the only SDS provider option supported in the Preview release. SPIRE is an Identity Management platform which at its heart is a tool chain that automatically issues and rotates authorized SVIDs (SPIFFE Verifiable Identity Document). A SPIRE Agent will run on each of the nodes on the cluster and will expose a Workload API via a Unix Domain Socket. All the envoys on a particular node will reach out to the local SPIRE Agent over UDS. Please refer [here](https://spiffe.io/docs/latest/spire/understand/) for additional details.
+
+In App Mesh, traffic encryption is enabled between Virtual Nodes and VirtualGateways, and thus between Envoys in your service mesh. This means your application code is not responsible for negotiating a TLS-encrypted session, instead allowing the local proxy(envoy) to negotiate and terminate TLS on your application's behalf. We will be configuring an SDS cluster in Envoy to obtain certificates from the SDS provider (i.e.,) SPIRE.
+
+## Prerequisites
+
+This feature is currently only available in [App Mesh preview](https://docs.aws.amazon.com/app-mesh/latest/userguide/preview.html) and will work with App Mesh controller [here](https://github.com/aws/eks-charts/tree/preview/stable/appmesh-controller). App Mesh preview is only provided in the `us-west-2` region.
+
+1. [Walkthrough: App Mesh with EKS](../eks/)
+2. Run the following to check the version of controller you are running.
+```
+$ kubectl get deployment -n appmesh-system appmesh-controller -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':'|tail -n1
+
+v1.2.1-mtls-preview
+```
+3. [Setup](https://docs.aws.amazon.com/app-mesh/latest/userguide/preview.html) AWS CLI to use preview channel
+```
+curl https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json \
+        -o $HOME/appmesh-preview-model.json
+aws configure add-model \
+    --service-name appmesh-preview \
+    --service-model file://$HOME/appmesh-preview-model.json
+```
+
+4. Install Docker. `deploy_app.sh` script builds the demo application images using Docker CLI.
+
+## Step 1: Setup Environment
+1. Clone this repository and navigate to the walkthrough/howto-k8s-mtls-sds-based folder, all commands will be ran from this location
+2. Your AWS account id:
+
+    export AWS_ACCOUNT_ID=<your_account_id>
+
+3. Region e.g. us-west-2
+
+    export AWS_DEFAULT_REGION=us-west-2
+
+4. ENVOY_IMAGE environment variable is set to App Mesh Envoy, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+
+    export ENVOY_IMAGE=...
```

Comment:
Should add a note below this that minimum envoy version `1.15.1.0` is required.

Reply:
Sure, will add it. Since, the above link is already pointing 1.15.1.0 that was left out but good to call it out.